### ISSUE TITLE
Odd bug with Python f-strings

### ISSUE THREAD

**Issue by kirbyfan64** [howl-editor/howl#379]
TL;DR: when using f-strings, the end of the string will sometimes randomly not be picked up. I think. I actually have no freaking clue.

Also, this is *really* hard to get a small test case for.

Basically, I open up [this file](https://github.com/kirbyfan64/sawanobot/blob/master/sawanobot.py) and scroll to line 184. It looks like this:

![howl_029](https://user-images.githubusercontent.com/1690697/30563787-9dce9a72-9c88-11e7-8b72-1c43ac4bcd0b.png)

HOWEVER, if I go to line 164, remove the `f`, and add it back, then the syntax highlighting goes back to normal.

This is probably something I did wrong when I added f-strings support to Python, but I'm not sure what. Same thing has happened when I added interpolation support to the Dart lexer...
